### ISSUE TITLE
Odd symbol comparison behaviour

### ISSUE THREAD

**Issue by 2bt** [leppie/IronScheme#6]
I've come across this behaviour which wasn't quite what I expected. Is it an actual bug or am I missing something?

```C#
Console.WriteLine("(eq? 'A 'A)".Eval());										// True
Console.WriteLine("(eq? 'A {0})".Eval("'A".Eval()));							// True
Console.WriteLine("(eq? 'A {0})".Eval((SymbolId) "'A".Eval()));					// False
Console.WriteLine("(eq? 'A {0})".Eval((SymbolId) "A"));							// False
Console.WriteLine("(eq? {0} {1})".Eval((SymbolId) "A", (SymbolId) "A"));		// False
Console.WriteLine("eq?".Eval<Callable>().Call((SymbolId) "A", (SymbolId) "A"));	// False
```

**Comment by leppie** [comment]
Thanks! Will check it out.

**Comment by leppie** [comment]
This is caused by boxing behavior in C#. All symbols in IronScheme are interned, hence a reference comparison works.

The best way to do this specific case is to emulate what the compiler does and use `SymbolTable.StringToObject(string)` or use `SymbolTable.Intern(SymbolId)`.

It might be a good idea though for me to check the parameters for `SymbolId` type and call that `Intern` for `Eval`. I will play with the idea.

**Comment by leppie** [comment]
Good news (and 1 bad).

I can get all of them to return `true` except for the last. 

I suggest you use the workaround with `SymbolTable` there.

Will make the change tonight and commit and rebuild.

**Comment by 2bt** [comment]
Thanks for the fast response and your suggested workaround.

**Comment by leppie** [comment]
Fixed in commit 2716cab72a
